File: PdfClown-NetStandard/CliSamples/org/pdfclown/samples/cli/AdvancedTextExtractionSample.cs
using org.pdfclown.documents;
using org.pdfclown.documents.contents;
using org.pdfclown.files;
using org.pdfclown.tools;

using System;
using System.Collections.Generic;
using System.Drawing;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to retrieve text content along with its graphic attributes
    (font, font size, text color, text rendering mode, text bounding box, and so on) from a PDF document;
    text is automatically sorted and aggregated.</summary>
  */
  public class AdvancedTextExtractionSample
    : Sample
  {
    public override void Run(
      )
    {
      // 1. Opening the PDF file...
      string filePath = PromptFileChoice("Please select a PDF file");
      using(var file = new File(filePath))
      {
        Document document = file.Document;
  
        // 2. Text extraction from the document pages.
        TextExtractor extractor = new TextExtractor();
        foreach(Page page in document.Pages)
        {
          if(!PromptNextPage(page, false))
          {
            Quit();
            break;
          }

          IList<ITextString> textStrings = extractor.Extract(page)[TextExtractor.DefaultArea];
          foreach(ITextString textString in textStrings)
          {
            RectangleF textStringBox = textString.Box.Value;
            Console.WriteLine(
              "Text ["
                + "x:" + Math.Round(textStringBox.X) + ","
                + "y:" + Math.Round(textStringBox.Y) + ","
                + "w:" + Math.Round(textStringBox.Width) + ","
                + "h:" + Math.Round(textStringBox.Height)
                + "]: " + textString.Text
                );
          }
        }
      }
    }
  }
}

File: java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AdvancedTextExtractionSample.java
package org.pdfclown.samples.cli;

import java.awt.geom.Rectangle2D;
import java.io.IOException;
import java.util.List;

import org.pdfclown.documents.Document;
import org.pdfclown.documents.Page;
import org.pdfclown.documents.contents.ITextString;
import org.pdfclown.files.File;
import org.pdfclown.tools.TextExtractor;

/**
  This sample demonstrates how to <b>retrieve text content along with its graphic attributes</b>
  (font, font size, text color, text rendering mode, text bounding box, and so on) from a PDF document;
  text is <i>automatically sorted and aggregated</i>.

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.0.8
  @version 0.1.2, 09/24/12
*/
public class AdvancedTextExtractionSample
  extends Sample
{
  @Override
  public void run(
    )
  {
    // 1. Opening the PDF file...
    File file;
    {
      String filePath = promptFileChoice("Please select a PDF file");
      try
      {file = new File(filePath);}
      catch(Exception e)
      {throw new RuntimeException(filePath + " file access error.",e);}
    }
    Document document = file.getDocument();

    // 2. Text extraction from the document pages.
    TextExtractor extractor = new TextExtractor();
    for(Page page : document.getPages())
    {
      if(!promptNextPage(page, false))
      {
        quit();
        break;
      }

      List<ITextString> textStrings = extractor.extract(page).get(null);
      for(ITextString textString : textStrings)
      {
        Rectangle2D textStringBox = textString.getBox();
        System.out.println(
          "Text ["
            + "x:" + Math.round(textStringBox.getX()) + ","
            + "y:" + Math.round(textStringBox.getY()) + ","
            + "w:" + Math.round(textStringBox.getWidth()) + ","
            + "h:" + Math.round(textStringBox.getHeight())
            + "]: " + textString.getText()
            );
      }
    }

    // 3. Closing the PDF file...
    try
    {file.close();}
    catch(IOException e)
    {throw new RuntimeException(file.getPath() + " file could not be closed.",e);}
  }
}